tests/wp-unit/include/Core/Metas/Modules/DeleteCommentMetaModuleTest.php

```diff
@@ -241,167 +241,689 @@ public function test_that_comment_meta_can_be_deleted_from_multiple_comments_wit
-						'meta_value' => 'Matched Value',
+					array(
+						'post_type'          => 'post',
+						'number_of_comments' => 5,
+						'matched'            => array(
+							'meta_key'   => 'test_key',
+							'meta_value' => 'Matched Value',
+						),
+						'miss_matched'       => array(
+							'meta_key'   => 'another_key',
+							'meta_value' => 'Matched Value',
+						),
 					),
-					'miss_matched'       => array(
-						'meta_key'   => 'test_key',
-						'meta_value' => 'Miss Matched Value',
+					array(
+						'post_type'          => 'post',
+						'number_of_comments' => 3,
+						'matched'            => array(
+							'meta_key'   => 'test_key',
+							'meta_value' => 'Test',
+						),
+						'miss_matched'       => array(
+							'meta_key'   => 'one_more_key',
+							'meta_value' => 'Value',
+						),
+					),
+					array(
+						'post_type'          => 'post',
+						'number_of_comments' => 3,
+						'miss_matched'       => array(
+							'meta_key'   => 'test_key',
+							'meta_value' => 'Matched Values',
+						),
 					),
 				),
 				array(
+					'post_type'  => 'post',
 					'meta_key'   => 'test_key',
 					'meta_type'  => 'CHAR',
-					'meta_value' => 'Matched Value',
-					'operator'   => '!=',
+					'meta_value' => 'Matched Value,Test',
+					'operator'   => 'IN',
 				),
 				array(
-					'number_of_comment_metas_deleted' => '5',
-					'explicit_meta_data'              => 'Matched Value',
+					'number_of_comment_metas_deleted' => 8,
 				),
 			),
+			array(
+				array(
+					array(
+						'post_type'          => 'post',
+						'number_of_comments' => 5,
+						'matched'            => array(
+							'meta_key'   => 'test_key',
+							'meta_value' => date( 'Y-m-d', strtotime( '-1 day' ) ),
+						),
+						'miss_matched'       => array(
+							'meta_key'   => 'another_key',
+							'meta_value' => date( 'Y-m-d', strtotime( '-1 day' ) ),
+						),
+					),
+					array(
+						'post_type'          => 'post',
+						'number_of_comments' => 3,
+						'miss_matched'       => array(
+							'meta_key'   => 'test_key',
+							'meta_value' => date( 'Y-m-d', strtotime( '-3 day' ) ),
+						),
+					),
+					array(
+						'post_type'          => 'post',
+						'number_of_comments' => 2,
+						'matched'            => array(
+							'meta_key'   => 'test_key',
+							'meta_value' => date( 'Y-m-d', strtotime( '-2 day' ) ),
+						),
+					),
+				),
+				array(
+					'post_type'  => 'post',
+					'meta_key'   => 'test_key',
+					'meta_type'  => 'DATE',
+					'meta_value' => date( 'Y-m-d', strtotime( '-1 day' ) ) . ',' . date( 'Y-m-d', strtotime( '-2 day' ) ),
+					'operator'   => 'IN',
+				),
+				array(
+					'number_of_comment_metas_deleted' => 7,
+				),
+			),
+		);
+	}
 
+	/**
+	 *  Data Provider for BETWEEN operator with numeric and date types.
+	 *
+	 * @return array Data
+	 */
+	public function provide_data_to_test_that_comment_meta_from_multiple_comments_can_be_deleted_using_value_with_between_operator() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method provide_data_to_test_that_comment_meta_from_multiple_comments_can_be_deleted_using_value_with_between_operator() has 126 lines of code. Current threshold is set to 100. Avoid really long methods.](https://app.codacy.com/app/BulkWP/bulk-delete/pullRequest?prid=2562677)